runtime/ftplugin/javascript.vim

```diff
@@ -26,13 +27,63 @@ setlocal comments=sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,://
 
 setlocal commentstring=//%s
 
-" Change the :browse e filter to primarily show Java-related files.
+" Change the :browse e filter to primarily show JavaScript-related files.
 if has("gui_win32")
-    let  b:browsefilter="Javascript Files (*.js)\t*.js\n" .
-		\	"All Files (*.*)\t*.*\n"
+    let  b:browsefilter="Javascript Files (*.js, *.jsx, *.es, *.es6, *.cjs, *.mjs, *.jsm, *.vue, *.json)\t*.js;*.jsx;*.es;*.es6;*.cjs;*.mjs;*.jsm;*.vue;*.json\n" .
+                \ "All Files (*.*)\t*.*\n"
 endif
-       
-let b:undo_ftplugin = "setl fo< ofu< com< cms<" 
+
+" The following suffixes should be implied when resolving filenames
+setlocal suffixesadd+=.js,.jsx,.es,.es6,.cjs,.mjs,.jsm,.vue,.json
+
+" The following suffixes should have low priority
+"   .snap    jest snapshot
+setlocal suffixes+=.snap
+
+" Prepend node_modules/.bin to $PATH if applicable
+" Allows calling executables installed locally via npm/yarn
+let s:bin_dir = finddir('node_modules/.bin', '.;')->fnamemodify(':p')
+if len(s:bin_dir) && $PATH !~ s:bin_dir
+    let $PATH = s:bin_dir .. ':' .. $PATH
+endif
+unlet s:bin_dir
+
+" Set 'path' to a minimal, non-greedy, value
+" User is expected to augment it with contextually-relevant paths
+setlocal path-=node_modules
+setlocal path-=/usr/include
+setlocal path-=**
+
+" Matchit configuration
+let b:match_words = '\<function\>:\<return\>,'
+            \ .. '\<do\>:\<while\>,'
+            \ .. '\<switch\>:\<case\>:\<default\>,'
+            \ .. '\<if\>:\<else\>,'
+            \ .. '\<try\>:\<catch\>:\<finally\>,'
+            \ .. '<\(\w\+\):</\1>'
+
+" Use eslint for :make if applicable
+if executable('eslint')
+    compiler eslint
+endif
+
+" Use standard for :make if applicable
+if executable('standard')
+    compiler standard
+endif
```

Comment:
- Including them here means less fiddling and a better out-of-the-box experience.
- Testing for `current_compiler` is already handled at the compiler level.

Reply:
> I don't feel strongly about it but that's the way it's done and I don't see a lot of value in departing from the project norms. However, if you were keen to offer a default it might be worth adding a javascript_default_compiler config variable like the Ada ftplugin does.

The problem, here, is that JavaScript is not a compiled language that comes with a default or recommended compiler. Defaulting to GNAT makes sense in Ada but defaulting to anything doesn't make sense in JS-land.

The right "compiler" is always project-specific and generally installed locally so having a default compiler, user-configurable or not, doesn't make much sense to me.

> It's being tested in a different context there. That's so a user compiler can block the default compiler from being loaded. The `:compiler` command actually deletes it when called from the ftplugin.

That seems strange but I'll take your word for this.

Reply:
> You are, however, currently favouring `eslint` over `standard` in this setup.

No. If you have both, `standard` is set last. If you only have `eslint`, `eslint` is set. And if you only have `standard`, `standard` is set. ESLint being a dependency of Standard, that behaviour makes perfect sense.

> I have at least a half dozen other compilers that I've been meaning to clean up and send upstream...for some years.

I already have pushed `eslint.vim` and `standard.vim`. What others do you have in mind? JSHint? Maybe Babel?

`:compiler` is not a special command. Like all its friends it can be used in mappings, other commands, scripts, autocommands, etc. Or in a ftplugin.